gap/oper.gi

```diff
@@ -1725,6 +1725,37 @@ function(D, list)
   return DigraphShortestDistance(D, list[1], list[2]);
 end);
 
+InstallMethod(VerticesReachableFrom, "for a digraph and a vertex",
+[IsDigraph, IsPosInt],
+function(D, root)
+  local N, index, current, succ, visited, prev, n, i, parent;
+  N := DigraphNrVertices(D);
+  index := ListWithIdenticalEntries(N, 1);
+  current := root;
+  succ := OutNeighbours(D);
+  visited := [root];
+  parent := [];
+  parent[root] := fail;
+  repeat
+    prev := current;
+    for i in [index[current] .. Length(succ[current])] do
+      n := succ[current][i];
+      if not (n in visited) then
+        Add(visited, n);
```

Comment:
The test in the `if` statement should be a lot faster if you do this:
```suggestion
      if not (n in visited) then
        AddSet(visited, n);
```
That's because a set is a *sorted* list in GAP, and GAP can use binary search with complexity `log n` to search in a sorted list of length `n`, otherwise it uses a linear search with complexity `n`.